ssl/quic/quic_impl.c

```diff
@@ -101,6 +104,26 @@ static ossl_inline int expect_quic_conn(const QUIC_CONNECTION *qc)
 
 }
 
+/*
+ * Ensures that the channel mutex is held for a method which touches channel
+ * state.
+ *
+ * Precondition: Channel mutex is not held (unchecked)
+ */
+static int quic_lock(QUIC_CONNECTION *qc)
+{
+    ossl_crypto_mutex_lock(qc->mutex);
+    return 1;
```

Comment:
Do we always need a mutex? What about the case where we're not using thread assisted mode?

Reply:
We could avoid it. I suspect this is not worth optimizing yet however as it will be uncontended in this case and should thus be very cheap. Moreover once we do multi-stream clients which make calls to different streams on multiple threads we will need it.